app/controllers/embeds/picos_controller.rb

```diff
@@ -0,0 +1,21 @@
+class Embeds::PicosController < ApplicationController
+  layout false
+
+  # Generates an embedable iframe in javascript format and a PICO map
+  # as html.
+  def show
+    respond_to do |format|
+      format.html do
+        set_scenario_area
+        @pico_module = params[:pico_module]
+      end
+    end
+  end
+
+  private
+
+  def set_scenario_area
+    scenario = Api::Scenario.find(Current.setting.api_session_id)
```

Comment:
I'm not sure if this is necessary. If you only want the area code, it should be available as `Current.setting.area_code` and will save the extra request to the API.

Reply:
I thought i tried that and failed. I might have done something wrong though. I'll retry.